src/js/test/rpc/server.test.ts

```diff
@@ -244,5 +244,79 @@ describe("Server", function () {
           });
       }
     );
+
+    it("should apply coprocessors to multiples record batches", function (done) {
+      const { spyGetHandles } = createStubs(sinonInstance);
+      // transform coprocessor function
+      const uppercase = (record) => ({
+        ...record,
+        value: record.value.map((char) => {
+          if (char > 97 && char < 122) {
+            return char - 32;
+          } else {
+            return char;
+          }
+        }),
+      });
+
+      const coprocessors = [1, 2, 3].map((id) =>
+        createHandle({
+          globalId: BigInt(id),
+          apply: (recordBatch) => {
+            const result = new Map();
+            const transformedRecord =
+              // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+              // @ts-ignore
+              recordBatch.map(({ header, records }) => {
+                return {
+                  header,
+                  records: records.map(uppercase),
+                };
+              });
+            result.set("result", transformedRecord);
+            return result;
+          },
+        })
+      );
+
+      spyGetHandles.returns(coprocessors);
+
+      Promise.all([
+        client.process_batch({
+          // send 101 record batches for apply 1,2,3 coprocessor
+          requests: "1"
+            .repeat(100)
+            .split("1")
+            .map(() => ({
+              recordBatch: [
+                createRecordBatch({
+                  header: {
+                    recordCount: 1,
+                  },
+                  records: [{ value: Buffer.from("b") }],
+                }),
+              ],
+              coprocessorIds: [BigInt(1), BigInt(2), BigInt(3)],
+              ntp: { partition: 1, namespace: "", topic: "produce" },
+            })),
+        }),
+      ])
+        .then(([{ result }]) => {
+          // 101 record batches * 3 coprocessor definition = 303
+          assert(result.length === 303);
+          result.forEach((processBatches) => {
+            processBatches.resultRecordBatch.forEach((rb) => {
+              rb.records.forEach((record) => {
+                // each record must have "A", because the coprocessors
+                // definition transform all letter to uppercase
+                assert.strictEqual(record.value.toString(), "B");
+                assert.strictEqual(record.valueLen, 1);
+              });
+            });
+          });
+          done();
```

Comment:
I 'think' that if you just return the promise you won't need to invoke 'done()'. Mocha is smart enough to know to wait on the future before the test ends, but you must return it or else it won't have a handle to the async action to wait on

Reply:
No, mocha isn't too smart hehe, if I don't put `done()` this test depends on finish < 200 ms that is possible but isn't recommend, mocha docs recommend use `done()` or `async/await`

Reply:
yeah isn't `async/await` promises? So just `return Promise.all( ... logic ... )`

Reply:
Oh if I use `async/await` I should change all promises implementation because we try to handle everything  with promise

Reply:
Done, it's working, thanks man